### PR TITLE
Created and reworked branch from 'action_group_api'

### PR DESCRIPTION
**Pull request by aabeshov**
### Description
Created and reworked branch from 'action_group_api'

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by aabeshov** [comment]
@Xtansia Hello, I can't solve problem that I can get through Build OpenAPI specs successfully.
This branch is synched with upstream/main branch, but when I add my api can't get pass test.

File: model/opensearch.smithy [hunk 1]
```diff
@@ -332,7 +338,6 @@ service OpenSearch {
         Search_Get_WithIndex,
         Search_Post,
         Search_Post_WithIndex,
-        SecurityHealth,
```
Comment (Xtansia) [inline model/opensearch.smithy:335]: This shouldn't be being removed here

File: model/security/action_group.smithy [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+//  The OpenSearch Contributors require contributions made to
+//  this file be licensed under the Apache-2.0 license or a
+//  compatible open source license.
+
+$version: "2"
+namespace OpenSearch
+
+structure CreateActionGroupParams{
+    allowed_actions: AllowedActions
+}
```
Comment (Xtansia) [inline model/security/action_group.smithy:12]: This is not necessary.
```suggestion
```

File: model/security/create_action_group/structures.smithy [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+$version: "2"
+namespace OpenSearch
+
+@input
+structure CreateActionGroup_Input {
+    @required
+    @httpLabel
+    action_group: String
+    @httpPayload
+    content: CreateActionGroupParams
```
Comment (Xtansia) [inline model/security/create_action_group/structures.smithy:16]: The payload for the create action is just the resource structure itself.
```suggestion
    content: Action_Group
```

File: model/security/patch_action_groups/structures.smithy [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    "x-version-added": "1.0",
+)
+@idempotent
+@suppress(["HttpUriConflict"])
+@http(method: "PATCH", uri: "/_plugins/_security/api/actiongroups")
+@documentation("Creates, updates, or deletes multiple action groups in a single call.")
+operation PatchActionGroups {
+    input: PatchActionGroups_Input,
+    output: PatchActionGroups_Output
+}
```
Comment (Xtansia) [inline model/security/patch_action_groups/structures.smithy:26]: This is the cause of your build errors, it should contain the `PatchActionGroups` Input and Output shapes:
```suggestion

@input
structure PatchActionGroups_Input {
    @httpPayload
    content: PatchOperationList
}

@output
structure PatchActionGroups_Output {
    status: MessageStatus,
    message: Message
}
```

**Review by Xtansia — CHANGES_REQUESTED** [review]
Thanks for recreating this PR @aabeshov, build errors are just because of a mixup with the input/output structures for the PatchActionGroups operation

**Comment by aabeshov** [comment]
@Xtansia Thank you very much for help.
Because when I was running ./gradlew build
Terminal gave me all the time some mistake with 'master_timeout' variables and I couldn't understand what's the problem.

**Review by Xtansia — APPROVED** [review]

**Review by VachaShah — APPROVED** [review]

**Pull request merged**